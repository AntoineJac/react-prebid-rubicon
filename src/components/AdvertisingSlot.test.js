import React from 'react';
import { spy, match } from 'sinon';
import expectSnapshot from '@mt-testutils/expect-snapshot';
import { mount } from 'enzyme';

const id = 'my-id';

describe('The advertising slot component', () => {
    let AdvertisingSlot, mockActivate;
    beforeEach(() => {
        mockActivate = spy();
        jest.mock('./utils/connectToAdServer', () => Component => props => (
            <Component {...props} activate={mockActivate} active={true} />
        ));
        AdvertisingSlot = require('./AdvertisingSlot').default;
    });
    it('renders correctly', () =>
        expectSnapshot(
            <AdvertisingSlot id={id} style={{ color: 'hotpink' }} className="my-class">
                <h1>hello</h1>
            </AdvertisingSlot>
        ));
    describe('when mounted', () => {
        beforeEach(() => mount(<AdvertisingSlot id={id} />));
        it('calls the activate function with the ID', () => void mockActivate.should.have.been.calledWith(id));
        it('calls the activate function with a collapse callback', () =>
            void mockActivate.should.have.been.calledWith(match.any, match.object));
    });
    afterEach(() => jest.resetModules());
});
