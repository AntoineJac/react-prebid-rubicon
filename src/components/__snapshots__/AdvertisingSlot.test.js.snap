// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The advertising slot component renders correctly 1`] = `
<div
  className="my-class"
  id="my-id"
  style={
    Object {
      "color": "hotpink",
    }
  }
>
  <h1>
    hello
  </h1>
</div>
`;
